src/main/java/com/fryrank/dal/ReviewDALImpl.java

```diff
@@ -36,20 +36,36 @@ public GetAllReviewsOutput getAllReviewsByRestaurantId(@NonNull final String res
         query.addCriteria(equalToRestaurantIdCriteria);
         final List<Review> reviews = mongoTemplate.find(query, Review.class);
 
-        final MatchOperation filterToRestaurantId = match(equalToRestaurantIdCriteria);
-        final GroupOperation averageScore = group(RESTAURANT_ID_KEY).avg("score").as("avgScore");
-        final Aggregation aggregation = newAggregation(filterToRestaurantId, averageScore);
-        final AggregationResults<RestaurantAvgScore> result = mongoTemplate.aggregate(aggregation, "review", RestaurantAvgScore.class);
-        final Optional<RestaurantAvgScore> uniqueResult = Optional.ofNullable(result.getUniqueMappedResult());
+        return new GetAllReviewsOutput(reviews);
+    }
+
+    @Override
+    public GetAggregateReviewInformationOutput getAggregateReviewInformationForRestaurants(@NonNull final List<String> restaurantIds, @NonNull final AggregateReviewFilter aggregateReviewFilter) {
+        final Query query = new Query().addCriteria(Criteria.where(RESTAURANT_ID_KEY).in(restaurantIds));
+
+        Map<String, AggregateReviewInformation> restaurantIdToAggregateReviewInformation = new HashMap<String, AggregateReviewInformation>();
 
-        if (uniqueResult.isPresent()) {
-            final Float rawScore = uniqueResult.get().getAvgScore();
-            final BigDecimal averageScoreBigDecimal = new BigDecimal(rawScore).setScale(1, RoundingMode.DOWN);
+        for(String restaurantId : restaurantIds) {
+            final Criteria includeRestaurantIdsCriteria = Criteria.where(RESTAURANT_ID_KEY).is(restaurantId);
+            final MatchOperation filterToRestaurantId = match(includeRestaurantIdsCriteria);
+            final GroupOperation averageScoreGroupOperation = group(RESTAURANT_ID_KEY).avg("score").as("avgScore");
+            final Aggregation aggregation = newAggregation(filterToRestaurantId, averageScoreGroupOperation);
+            final AggregationResults<RestaurantAvgScore> result = mongoTemplate.aggregate(aggregation, "review", RestaurantAvgScore.class);
+            final Optional<RestaurantAvgScore> uniqueResult = Optional.ofNullable(result.getUniqueMappedResult());
 
-            return new GetAllReviewsOutput(reviews, averageScoreBigDecimal.floatValue());
-        } else {
-            return new GetAllReviewsOutput(reviews, null);
+            if (uniqueResult.isPresent()) {
+                Float averageScore = null;
```

Comment:
could make this Optional.empty rather than null to more verbosely communicate the intention of having a value or not

Reply:
Opted against making the change for now; changing the output value to Optional requires additional refactoring and is probably unnecessary